### PR TITLE
Superlinter configuration

### PR DESCRIPTION
**Pull request by jdziurzynski**
KNTK-323

### CODE REVIEW

File: cloud_AWS/terraform/module/examples/multiple-aws-accounts-multiple-vpc-setup/aws_onboarder.py [hunk 1]
```diff
@@ -23,8 +23,12 @@ class AwsProfile:
     secret_key: str
 
 
-TerraformAction = Callable[[Terraform, str], bool]  # terraform plan/apply/destroy for specified region
-RequestedProfileNames = Optional[List[str]]  # list of profile names matching profiles in ~/.aws/credentials
```
Comment (mateuszmidor) [inline cloud_AWS/terraform/module/examples/multiple-aws-accounts-multiple-vpc-setup/aws_onboarder.py:27]: Hmm why does this line break, it is under 120 chars long?

**Review by mateuszmidor — CHANGES_REQUESTED** [review]
Most of the changes originate from breaking the lines in python code, these lines fit in 120 characters limit so there must be something wrong with code formatter config; please investigate and bring back order :)

File: cloud_AWS/terraform/module/examples/multiple-aws-accounts-multiple-vpc-setup/aws_onboarder.py (hunk 1 above)
Comment (mmac-m3a) [reply cloud_AWS/terraform/module/examples/multiple-aws-accounts-multiple-vpc-setup/aws_onboarder.py:27]: Looks like the linter doing the line-breaking is configured to 100 chars per line. This should be fixed. We want 120 chars per line.

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
Line length should be limited to 120 chars in Python code (not 100 as it appears to be right now). Looks like the `flake8` setting is not being picked up  or the line-wrapping is done by some other tool than `flake8`.

File: .github/linters/.jscpd.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "threshold": 0,
+  "reporters": [
+    "consoleFull"
+  ],
+  "ignore": [
+    "**/__snapshots__/**",
+    "**/cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/**",
```
Comment (mateuszmidor) [inline .github/linters/.jscpd.json:8]: why ignore?

File: .github/linters/.markdown-lint.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# - https://github.com/DavidAnson/markdownlint
+
+###############
+# Rules by id #
+###############
+MD004: false                  # Unordered list style
+MD007:
+  indent: 2                   # Unordered list indentation
+MD013:
+  line_length: 400            # Line length 80 is far to short
```
Comment (mateuszmidor) [inline .github/linters/.markdown-lint.yml:12]: "too short"

Comment (jdziurzynski) [reply .github/linters/.markdown-lint.yml:12]: Most of this configuration comes from the default github superlinter configuration file. I have added some checks to ignore.
Line length I left with a default value. 
When I view .md documents on github/IDE everything looks fine. With a line length of 400, we have no problem with formatting the tables.

**Review by mateuszmidor — APPROVED** [review]
Looks good

File: .github/linters/.jscpd.json (hunk 2 above)
Comment (jdziurzynski) [reply .github/linters/.jscpd.json:8]: [Here](https://github.com/kentik/config-snippets-cloud/runs/6905360928?check_suite_focus=true#step:4:24379) and [here](https://github.com/kentik/config-snippets-cloud/runs/6905360928?check_suite_focus=true#step:4:24906) You can find errors about it. 
I wouldn't change demo files at all. 
To fix this linter check in python files would need a small refactor of this scripts, so I decided to leave it as it is.

File: .github/linters/.markdown-lint.yml (hunk 3 above)
Comment (mateuszmidor) [reply .github/linters/.markdown-lint.yml:12]: I meant the comment: Line length 80 is **to** short
:)

**Comment by mmac-m3a** [comment]
Replaced by a more conservative PR.

**Pull request closed without merging**